### ISSUE TITLE
Unable to install yandex-cloud-cli -  docker-credential-yc is not there

### ISSUE THREAD

**Issue by GeckoSplinter** [Homebrew/homebrew-cask#194920]
### Verification

- [X] I understand that [if I ignore these instructions, my issue may be closed without review](https://github.com/Homebrew/homebrew-cask/blob/master/doc/faq/closing_issues_without_review.md).
- [X] I have retried my command with `--force`.
- [X] I ran `brew update-reset && brew update` and retried my command.
- [X] I ran `brew doctor`, fixed as many issues as possible and retried my command.
- [X] I have checked the instructions for [reporting bugs](https://github.com/Homebrew/homebrew-cask#reporting-bugs).
- [X] I made doubly sure this is not a [checksum does not match / SHA256 mismatch](https://docs.brew.sh/Common-Issues#cask---checksum-does-not-match) error (do not open an issue before trying to open a PR to fix first).

### Description of issue

I am trying to install the yandex-cli with homebrew and I got this issue.

Looking at the install script from yandex, the docker-credentials-yc is not anymore provided.

But that's not all, if I fix that part in the cask I am hitting another issue because I use fish shell, so bash and zsh completion are not installed. 
And anyway if you are using bash only bash completion will be installed and same with zsh, so the binary stanza will failed.

```
BASH_COMPLETION_AVAILABLE=
if [ "${SHELL_NAME}" = "bash" ]; then
    BASH_COMPLETION_AVAILABLE=yes
fi

...

if [ "${BASH_COMPLETION_AVAILABLE}" = "yes" ]; then
    "${CLI_BIN_FULL_PATH}" completion bash > "${CLI_BASH_COMPLETION}"
fi
```

Same goes with ZSH

I will open a PR removing those check if it makes sense. 

```
brew install yandex-cloud-cli
.....
Error: It seems the symlink source '/opt/homebrew/Caskroom/yandex-cloud-cli/0.140.0/yandex-cloud-cli/bin/docker-credential-yc' is not there.```


### Command that failed

brew install yandex-cloud-cli

### Output of command with `--verbose --debug`

```shell
==> Downloading https://formulae.brew.sh/api/formula.jws.json
####################################################################################################################################################################################### 100.0%
==> Downloading https://formulae.brew.sh/api/cask.jws.json
####################################################################################################################################################################################### 100.0%
==> Caveats
To install shell completions add this to your profile:

  for bash users
    source "/opt/homebrew/Caskroom/yandex-cloud-cli/0.140.0/yandex-cloud-cli/completion.bash.inc"

  for zsh users
    source "/opt/homebrew/Caskroom/yandex-cloud-cli/0.140.0/yandex-cloud-cli/completion.zsh.inc"

==> Downloading https://storage.yandexcloud.net/yandexcloud-yc/install.sh
Already downloaded: /Users/thomas.perronin/Library/Caches/Homebrew/downloads/e0e68b066c559c1f3270ca5dda658681b4ba835addcd465d99f828ce47298c55--install.sh
Warning: No checksum defined for cask 'yandex-cloud-cli', skipping verification.
==> Installing Cask yandex-cloud-cli
==> Running installer script 'install.sh'
Downloading yc 0.140.0
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 95.6M  100 95.6M    0     0  21.0M      0  0:00:04  0:00:04 --:--:-- 23.7M
Yandex Cloud CLI 0.140.0 darwin/arm64
yc is installed to /opt/homebrew/Caskroom/yandex-cloud-cli/0.140.0/yandex-cloud-cli/bin/yc
==> Purging files for version 0.140.0 of Cask yandex-cloud-cli
Error: It seems the symlink source '/opt/homebrew/Caskroom/yandex-cloud-cli/0.140.0/yandex-cloud-cli/bin/docker-credential-yc' is not there.

☸ preprod-1 in ~ on ☁️  default took 9s
❯ brew install yandex-cloud-cli --verbose --debug
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::NullLoader): loading yandex-cloud-cli
/opt/homebrew/Library/Homebrew/brew.rb (Formulary::NullLoader): loading yandex-cloud-cli
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading yandex-cloud-cli
==> Cask::Installer#install
==> Printing caveats
==> Caveats
To install shell completions add this to your profile:

  for bash users
    source "/opt/homebrew/Caskroom/yandex-cloud-cli/0.140.0/yandex-cloud-cli/completion.bash.inc"

  for zsh users
    source "/opt/homebrew/Caskroom/yandex-cloud-cli/0.140.0/yandex-cloud-cli/completion.zsh.inc"

==> Cask::Installer#fetch
==> Downloading https://storage.yandexcloud.net/yandexcloud-yc/install.sh
/usr/bin/env /opt/homebrew/Library/Homebrew/shims/shared/curl --disable --cookie /dev/null --globoff --show-error --user-agent Homebrew/4.4.11\ \(Macintosh\;\ arm64\ Mac\ OS\ X\ 15.1.1\)\ curl/8.7.1 --header Accept-Language:\ en --retry 3 -V
/usr/bin/env /opt/homebrew/Library/Homebrew/shims/shared/curl --disable --cookie /dev/null --globoff --show-error --user-agent Homebrew/4.4.11\ \(Macintosh\;\ arm64\ Mac\ OS\ X\ 15.1.1\)\ curl/8.7.1 --header Accept-Language:\ en --retry 3 --fail --location --silent --head https://storage.yandexcloud.net/yandexcloud-yc/install.sh
/usr/bin/env /opt/homebrew/Library/Homebrew/shims/shared/curl --disable --cookie /dev/null --globoff --show-error --user-agent Homebrew/4.4.11\ \(Macintosh\;\ arm64\ Mac\ OS\ X\ 15.1.1\)\ curl/8.7.1 --header Accept-Language:\ en --retry 3 --fail --location --silent --head --request GET --http1.1 https://storage.yandexcloud.net/yandexcloud-yc/install.sh

Already downloaded: /Users/thomas.perronin/Library/Caches/Homebrew/downloads/e0e68b066c559c1f3270ca5dda658681b4ba835addcd465d99f828ce47298c55--install.sh
==> Checking quarantine support
/usr/bin/env /usr/bin/xattr -h
/usr/bin/env /Library/Developer/CommandLineTools/usr/bin/swift -target arm64-apple-macosx15 /opt/homebrew/Library/Homebrew/cask/utils/quarantine.swift
==> Quarantine is available.
==> Verifying Gatekeeper status of /Users/thomas.perronin/Library/Caches/Homebrew/downloads/e0e68b066c559c1f3270ca5dda658681b4ba835addcd465d99f828ce47298c55--install.sh
/usr/bin/env /usr/bin/xattr -p com.apple.quarantine /Users/thomas.perronin/Library/Caches/Homebrew/downloads/e0e68b066c559c1f3270ca5dda658681b4ba835addcd465d99f828ce47298c55--install.sh
==> /Users/thomas.perronin/Library/Caches/Homebrew/downloads/e0e68b066c559c1f3270ca5dda658681b4ba835addcd465d99f828ce47298c55--install.sh is quarantined
Warning: No checksum defined for cask 'yandex-cloud-cli', skipping verification.
/usr/bin/env hdiutil imageinfo -format /Users/thomas.perronin/Library/Caches/Homebrew/downloads/e0e68b066c559c1f3270ca5dda658681b4ba835addcd465d99f828ce47298c55--install.sh
==> Installing Cask yandex-cloud-cli
==> Cask::Installer#stage
==> Extracting primary container
==> Using container class UnpackStrategy::Executable for /Users/thomas.perronin/Library/Caches/Homebrew/downloads/e0e68b066c559c1f3270ca5dda658681b4ba835addcd465d99f828ce47298c55--install.sh
cp -p /Users/thomas.perronin/Library/Caches/Homebrew/downloads/e0e68b066c559c1f3270ca5dda658681b4ba835addcd465d99f828ce47298c55--install.sh /opt/homebrew/Caskroom/yandex-cloud-cli/0.140.0/install.sh
==> Verifying Gatekeeper status of /Users/thomas.perronin/Library/Caches/Homebrew/downloads/e0e68b066c559c1f3270ca5dda658681b4ba835addcd465d99f828ce47298c55--install.sh
/usr/bin/env /usr/bin/xattr -p com.apple.quarantine /Users/thomas.perronin/Library/Caches/Homebrew/downloads/e0e68b066c559c1f3270ca5dda658681b4ba835addcd465d99f828ce47298c55--install.sh
==> /Users/thomas.perronin/Library/Caches/Homebrew/downloads/e0e68b066c559c1f3270ca5dda658681b4ba835addcd465d99f828ce47298c55--install.sh is quarantined
==> Propagating quarantine from /Users/thomas.perronin/Library/Caches/Homebrew/downloads/e0e68b066c559c1f3270ca5dda658681b4ba835addcd465d99f828ce47298c55--install.sh to /opt/homebrew/Caskroom/yandex-cloud-cli/0.140.0
/usr/bin/env /usr/bin/xattr -p com.apple.quarantine /Users/thomas.perronin/Library/Caches/Homebrew/downloads/e0e68b066c559c1f3270ca5dda658681b4ba835addcd465d99f828ce47298c55--install.sh
/usr/bin/env /usr/bin/xargs -0 -- /bin/chmod -h u+w
/usr/bin/env /usr/bin/xargs -0 -- /usr/bin/xattr -w com.apple.quarantine 0381\;675972e3\;\;2DD2F67E-CA6A-4190-B8F5-C28387AA6F27
==> Creating metadata directory: /opt/homebrew/Caskroom/yandex-cloud-cli/.metadata/0.140.0/20241211114541.724
==> Creating metadata subdirectory: /opt/homebrew/Caskroom/yandex-cloud-cli/.metadata/0.140.0/20241211114541.724/Casks
==> Installing artifacts
==> Installing artifact of class Cask::Artifact::Installer
==> Running installer script 'install.sh'
/usr/bin/env PATH=/opt/homebrew/bin:/opt/homebrew/sbin:/opt/homebrew/Library/Homebrew/shims/shared:/usr/bin:/bin:/usr/sbin:/sbin /opt/homebrew/Caskroom/yandex-cloud-cli/0.140.0/install.sh -i /opt/homebrew/Caskroom/yandex-cloud-cli/0.140.0/yandex-cloud-cli -r /dev/null
Downloading yc 0.140.0
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 95.6M  100 95.6M    0     0  21.4M      0  0:00:04  0:00:04 --:--:-- 21.4M
Yandex Cloud CLI 0.140.0 darwin/arm64
yc is installed to /opt/homebrew/Caskroom/yandex-cloud-cli/0.140.0/yandex-cloud-cli/bin/yc
==> Installing artifact of class Cask::Artifact::Binary
==> Purging files for version 0.140.0 of Cask yandex-cloud-cli
Error: It seems the symlink source '/opt/homebrew/Caskroom/yandex-cloud-cli/0.140.0/yandex-cloud-cli/bin/docker-credential-yc' is not there.
/opt/homebrew/Library/Homebrew/cask/artifact/symlinked.rb:46:in `link'
/opt/homebrew/Library/Homebrew/cask/artifact/binary.rb:11:in `link'
/opt/homebrew/Library/Homebrew/cask/artifact/symlinked.rb:21:in `install_phase'
/opt/homebrew/Library/Homebrew/cask/installer.rb:255:in `block in install_artifacts'
/opt/homebrew/Library/Homebrew/cask/artifact_set.rb:10:in `each'
/opt/homebrew/Library/Homebrew/cask/artifact_set.rb:10:in `each'
/opt/homebrew/Library/Homebrew/cask/installer.rb:248:in `install_artifacts'
/opt/homebrew/Library/Homebrew/cask/installer.rb:115:in `install'
/opt/homebrew/Library/Homebrew/cmd/install.rb:245:in `block in run'
/opt/homebrew/Library/Homebrew/cmd/install.rb:234:in `each'
/opt/homebrew/Library/Homebrew/cmd/install.rb:234:in `run'
/opt/homebrew/Library/Homebrew/brew.rb:94:in `<main>'
```

### Output of `brew doctor` and `brew config`

```shell
brew doctor
Your system is ready to brew.

brew config
HOMEBREW_VERSION: 4.4.11
ORIGIN: https://github.com/Homebrew/brew
HEAD: ad356d365837a691a52dfec32b3fb5f282f3aefc
Last commit: 2 days ago
Branch: stable
Core tap JSON: 11 Dec 11:42 UTC
Core cask tap JSON: 11 Dec 11:42 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: nvim
HOMEBREW_MAKE_JOBS: 11
Homebrew Ruby: 3.3.6 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.6/bin/ruby
CPU: 11-core 64-bit arm_lobos
Clang: 16.0.0 build 1600
Git: 2.47.1 => /opt/homebrew/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.1.1-arm64
CLT: 16.1.0.0.1.1729049160
Xcode: N/A
Rosetta 2: false
```

### Output of `brew tap`

```shell
nikitabobko/tap
```

**Comment by daeho-ro** [comment]
The PR is merged and now will be fine.
